diff --git Makefile.am Makefile.am
index 2964abd..a5c791e 100644
--- Makefile.am
+++ Makefile.am
@@ -13,7 +13,6 @@ remove_bindistdir = \
          && rm -fr $(bindistdir); }; }
 
 install-exec-hook:
-	$(INSTALL) $(SUPPORT_BIN) $(bindir)
 	@cat support-files/install-exec-hook.txt
 
 dist-doc:
diff --git configure.in configure.in
index ebf4678..aeada6f 100644
--- configure.in
+++ configure.in
@@ -184,10 +184,22 @@ then
 fi
 AC_SUBST(BINARY_DIST_DIR)
 
+AC_PATH_PROG(MYSQLCFG, mysql_config, no, [/usr/bin /usr/local/bin])
+if test "$MYSQLCFG" = no; then
+    AC_MSG_ERROR([mysql_config not found.])
+else
+    MYSQL_INCLUDES="`$MYSQLCFG --include`"
+    MYSQL_LIBS="`$MYSQLCFG --libs`"
+    MYSQL_PLUGINDIR="`$MYSQLCFG --plugindir`"
+    AC_SUBST(MYSQL_INCLUDES)
+    AC_SUBST(MYSQL_LIBS)
+    AC_SUBST(MYSQL_PLUGINDIR)
+fi
+
 AC_C_CONST
 AC_TYPE_SIZE_T
 AC_CHECK_FUNCS([lseek64])
 AC_CHECK_FUNC(powl, [], [AC_DEFINE([powl],[pow],[use pow instead of powl])])
 AC_CHECK_HEADERS(limits.h syslimits.h)
 AC_SEARCH_LIBS([pthread_mutex_timedlock], [pthread], [AC_DEFINE([HAVE_PTHREAD_MUTEX_TIMEDLOCK],1,[use pthread_mutex_timedlock])])
-AC_OUTPUT(Makefile src/Makefile t/Makefile)
+AC_OUTPUT(Makefile src/Makefile t/Makefile support-files/Makefile)
diff --git src/Makefile.am src/Makefile.am
index 5134472..dad1d2c 100644
--- src/Makefile.am
+++ src/Makefile.am
@@ -1,18 +1,18 @@
-INCLUDES=			$(MYSQL_INC)
+INCLUDES=			$(MYSQL_INC) @MYSQL_INCLUDES@
 lib_LTLIBRARIES=		libqueue_engine.la
 noinst_HEADER=			ha_queue.h queue_cond.h dllist.h
 noinst_LIBRARIES=		libqueue_cond.a
 EXTRA_DIST=			adler32.c $(noinst_HEADER)
 
 libqueue_cond_a_SOURCES=	queue_cond.cc
-libqueue_cond_a_CFLAGS=		$(AM_CFLAGS) -fPIC
-libqueue_cond_a_CXXFLAGS=	$(AM_CFLAGS) -fPIC
+libqueue_cond_a_CFLAGS=		$(AM_CFLAGS) @MYSQL_INCLUDES@ -fPIC
+libqueue_cond_a_CXXFLAGS=	$(AM_CFLAGS) @MYSQL_LIBS@ -fPIC
 
 libqueue_engine_la_SOURCES=	ha_queue.cc
 libqueue_engine_la_LIBADD=	queue_cond.lo
-libqueue_engine_la_LDFLAGS=	-module
-libqueue_engine_la_CFLAGS=	$(AM_CFLAGS) -DMYSQL_DYNAMIC_PLUGIN
-libqueue_engine_la_CXXFLAGS=	$(AM_CFLAGS) -DMYSQL_DYNAMIC_PLUGIN -fno-rtti -fno-exceptions
+libqueue_engine_la_LDFLAGS=	-module  @MYSQL_LIBS@
+libqueue_engine_la_CFLAGS=	$(AM_CFLAGS) @MYSQL_INCLUDES@ -DMYSQL_DYNAMIC_PLUGIN
+libqueue_engine_la_CXXFLAGS=	$(AM_CFLAGS) @MYSQL_LIBS@ -DMYSQL_DYNAMIC_PLUGIN -fno-rtti -fno-exceptions
 
 dist-hook: dist-boost dist-cac
 
diff --git support-files/Makefile.am support-files/Makefile.am
new file mode 100644
index 0000000..b13459f
--- /dev/null
+++ support-files/Makefile.am
@@ -0,0 +1 @@
+bin_SCRIPTS = q4m-forward
diff --git t/Makefile.am t/Makefile.am
index 6bcb058..d98a0a0 100644
--- t/Makefile.am
+++ t/Makefile.am
@@ -1,15 +1,15 @@
 EXTRA_DIST=	*.t
 noinst_PROGRAMS=	05-multireader-read 05-multirw-core 05-multiwait-core
 
-INCLUDES=	$(MYSQL_INC)
-AM_LDFLAGS=	-L$(MYSQL_LIBDIR)
+INCLUDES=	$(MYSQL_INC)  @MYSQL_INCLUDES@
+AM_LDFLAGS=	-L$(MYSQL_LIBDIR) @MYSQL_LIBS@
 LDADD=		-lmysqlclient
 
 05_multireader_read_SOURCES=	05-multireader-read.c
-05_multireader_read_CFLAGS=	$(AM_CFLAGS)
+05_multireader_read_CFLAGS=	$(AM_CFLAGS) @MYSQL_INCLUDES@
 
 05_multirw_core_SOURCES=	05-multirw-core.c
-05_multirw_core_CFLAGS=		$(AM_CFLAGS)
+05_multirw_core_CFLAGS=		$(AM_CFLAGS) @MYSQL_INCLUDES@
 
 05_multiwait_core_SOURCES=	05-multiwait-core.c
-05_multiwait_core_CFLAGS=	$(AM_CFLAGS)
+05_multiwait_core_CFLAGS=	$(AM_CFLAGS) @MYSQL_INCLUDES@
